src/Common/src/System/Net/Http/TlsCertificateExtensions.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Defines Extension methods which are meant to be re-used across WinHttp & UnixHttp Handlers
+    /// </summary>
+    internal static class TLSCertificateExtensions
+    {
+        private const string ClientCertificateOid = "1.3.6.1.5.5.7.3.2";
+
+        /// <summary>
+        ///   returns true if the X509 Certificate can be used  as SSL Client Certificate.
+        /// </summary>
+        private static bool IsClientCertificate(X509Certificate2 cert)
+        {
+            Debug.Assert(cert != null, "certificate cannot be null");
+
+            bool foundEku = false;
+            bool foundKeyUsages = false;
+            bool isClientAuth = true;
+            bool isDigitalSignature = true;
+            foreach(X509Extension extension in cert.Extensions)
+            {
+                // check if the extension is an enhanced usage ext.
+                // But do this only if needed. No point going over it, if we already have established that our cert has the
+                // required extension.
+                if (!isClientAuth)
+                {
+                    X509EnhancedKeyUsageExtension enhancedUsageExt = extension as X509EnhancedKeyUsageExtension;
+                    if (enhancedUsageExt != null)
+                    {
+                        foundEku = true;
```

Comment:
You need to set IsClientAuth to false here, as well; because you already started it off as true.

true (default)
false (EKU extension is present)
true (client OID is present within the EKU extension)